redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt

```diff
@@ -126,85 +127,95 @@ internal class UIViewBox :
       super.layoutSubviews()
 
       children.widgets.forEach { widget ->
-        val view = widget.value
-        view.setFrame(frame)
-        view.sizeToFit()
+        layoutWidget(widget)
+      }
+    }
 
-        // Check for modifier overrides in the children, otherwise default to the Box's alignment values.
-        var itemHorizontalAlignment = horizontalAlignment
-        var itemVerticalAlignment = verticalAlignment
+    private fun layoutWidget(widget: Widget<UIView>) {
+      val view = widget.value
 
-        var requestedWidth: CGFloat = Double.MIN_VALUE
-        var requestedHeight: CGFloat = Double.MIN_VALUE
+      // Check for modifier overrides in the children, otherwise default to the Box's alignment values.
+      var itemHorizontalAlignment = horizontalAlignment
+      var itemVerticalAlignment = verticalAlignment
 
-        widget.modifier.forEachScoped { childModifier ->
-          when (childModifier) {
-            is HorizontalAlignment -> {
-              itemHorizontalAlignment = childModifier.alignment
-            }
+      val frameWidth = frame.useContents { size.width }
+      val frameHeight = frame.useContents { size.height }
 
-            is VerticalAlignment -> {
-              itemVerticalAlignment = childModifier.alignment
-            }
+      var requestedWidth: CGFloat = Double.NaN
+      var requestedHeight: CGFloat = Double.NaN
```

Comment:
Switched from Double.MIN_VALUE (which is a possible actual value near zero) to Double.NaN which isn't a possible actual value.